### PR TITLE
Bug 17097/diff tester does not clean output properly

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17097

### CODE REVIEW

File: rudder-lang/tools/cfjson_tester [hunk 1]
```diff
@@ -14,70 +14,113 @@ import configparser
...
-      empty_dict_cleaner(diff_dict[element])
-      if len(diff_dict[element]) == 0:
+    for element in diff_dict.copy():
+      cur = diff_dict[element]
+      empty_dict_cleaner(cur)
+      if len(cur) == 0:
         diff_dict.pop(element)
         return
+      elif len(cur) == 1 and type(cur) is dict and 'line' in cur.keys():
+        diff_dict.pop(element)
```
Comment (gpoblon) [inline rudder-lang/tools/cfjson_tester:122]: These two lines are the fix. 
The diff tool adds a lot of orphan lines dict, and when deleting "on purpose diffs" these "lines" remain and must be cleaned.

Comment (peckpeck) [reply rudder-lang/tools/cfjson_tester:126]: the type(cur) check should come first since python does lazy evaluation and len isn't available for every ype

**Comment by peckpeck** [comment]
@gpoblon this PR should be rebased

File: rudder-lang/tools/cfjson_tester (hunk 1 above)
Comment (gpoblon) [reply rudder-lang/tools/cfjson_tester:126]: ok fixing

**Comment by gpoblon** [comment]
can be merged

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**